cpiface/zmq-cpiface.cc

```diff
@@ -142,17 +146,56 @@ void sig_handler(int signo) {
   google::protobuf::ShutdownProtobufLibrary();
 }
 /*--------------------------------------------------------------------------------*/
+void force_restart(int argc, char **argv) {
+  pid_t pid;
+  int status;
+
+  pid = fork();
+
+  if (pid == -1) {
+    std::cerr << "Failed to fork: " << strerror(errno) << std::endl;
+    exit(EXIT_FAILURE);
+  } else if (pid == 0) {  // child process
+    execv(argv[0], argv);
+    exit(EXIT_SUCCESS);
+  } else {  // parent process
+    if (waitpid(pid, &status, 0) > 0) {
```

Comment:
So child is new cpiface and parent (old cpiface) dies eventually. Is that right ?

Reply:
Yes.